### PR TITLE
Chainer２系の対応

### PR DESCRIPTION
**Pull request by wararaki**
以下２点を修正
・Chainer 2系で動作するように、skchainerの各インターフェースを修正
・未使用のライブラリを削除（commonml/sklearn関連）

Chainer１系の互換性なし。

### CODE REVIEW

File: commonml/skchainer/classifier.py [hunk 1]
```diff
@@ -24,20 +24,17 @@ def __init__(self, predictor, lossfun, accfun,
...
+    def __call__(self, x, t):
         self.y = None
         self.loss = None
         self.accuracy = None
-        if self.has_train:
-            y = self.predictor(x, train=train)
-        else:
-            y = self.predictor(x)
-        self.loss = self.lossfun(y, t)
+        with chainer.using_config('train', True):
```
Comment (marevol) [inline commonml/skchainer/classifier.py]: 不要だと思います。

File: commonml/skchainer/classifier.py [hunk 2]
```diff
@@ -24,20 +24,17 @@ def __init__(self, predictor, lossfun, accfun,
...
         self.loss = None
         self.accuracy = None
-        self.has_train = 'train' in inspect.getargspec(self.predictor.__call__).args
         self.prefit_y = prefit_y
         self.astype_y = astype_y
         self.postpredict_y = postpredict_y
 
-    def __call__(self, x, t, train=True):
+    def __call__(self, x, t):
         self.y = None
```
Comment (marevol) [inline commonml/skchainer/classifier.py]: self.y, self.loss, self.accuracyが不要かもしれません。

File: commonml/skchainer/estimator.py [hunk 3]
```diff
@@ -98,38 +98,35 @@ def predict(self, X,
...
             dataset = dataset_creator(X)
 
-        has_train = 'train' in inspect.getargspec(self.model.predictor.__call__).args
 
         def predict_on_predictor(X):
-            if has_train:
-                return self.model.predictor(X, train=False)
-            else:
-                return self.model.predictor(X)
+            return self.model.predictor(X)
```
Comment (marevol) [inline commonml/skchainer/estimator.py]: predict_on_predictorをなくして、self.model.predictorを直接呼ぶので良いと思います。

File: commonml/skchainer/regressor.py [hunk 4]
```diff
@@ -24,18 +24,15 @@ def __init__(self, predictor, lossfun, accfun,
...
 
-    def __call__(self, x, t, train=True):
+    def __call__(self, x, t):
         self.y = None
         self.loss = None
         self.accuracy = None
-        if self.has_train:
-            self.y = self.predictor(x, train=train)
-        else:
+        with chainer.using_config('train', True):
```
Comment (marevol) [inline commonml/skchainer/regressor.py]: 不要だと思います。
